var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -195,6 +197,10 @@ class Petsc(Package, CudaPackage, ROCmPackage):
     # older versions of petsc did not support mumps when +int64
     conflicts('+mumps', when='@:3.12+int64')
 
+    # conflicts with ~fortran, these packages require +fortran
+    conflicts('+scalapack', when='~fortran')
+    conflicts('+superlu-dist', when='~fortran')
```

Comment:
You can add `when='+fortran'` to each of the two variants, and avoid the conflicts.

Reply:
The maintainer of one of those packages (superlu-dist) has been notified of that suggestion in an earlier thread!

Reply:
Hmm I don't fully understand the reply. I was suggesting:
```python
variant('scalapack', default=False, when='+fortran', description='Activates support for Scalapack')
```
and avoiding the conflicts. I didn't mean to advise to add a `fortran` variant to dependencies, if that's what you are referring to.

Reply:
I misunderstood the suggestion. I have now incorporated this in the latest commit.

Reply:
Thanks for clarifying that, I didn't know that conditional variants were now available in spack!

Reply:
If I understood the suggestion correctly - the `conflict` lines can now be removed.

Reply:
Done. Thanks for catching that!